js/ssai_pulse.js

```diff
@@ -602,6 +630,16 @@ OO.Ads.manager(function(_, $)
       }
     };
 
+    /**
+     * Called if the ajax call for SSAI metadata fails
+     * @public
+     * @method SsaiPulse#onMetadataError
+     */
+    this.onMetadataError = function(error)
+    {
+      OO.log("SSAI Metadata Request: Error" + JSON.stringify(error));
```

Comment:
What do we normally send when metadata fails in the core?  Should we send the same type of message here to the core, so IQ can record it?

Reply:
@pilievOoyala right now there is no specific message we use for ad metadata failure. This technically isn't a VAST request so it is not a request error. API_ERROR might work but I would need to add the ability to send that in the AMC. Also note that none of these are reported by reporter.js currently, they are only processed by the JSON pipeline

Reply:
@pilievOoyala added a function to let amc raise an api error